cmlreaders/base_reader.py

```diff
@@ -0,0 +1,87 @@
+from typing import Optional
+
+from .path_finder import PathFinder
+from .exc import UnsupportedOutputFormat
+from .cmlreader import CMLReader
+
+
+class MetaReader(type):
+    """ Metaclass for all CML readers
+
+        The responsibility of the metaclass is to register the reader with
+        the generic CMLReader to avoid needing to do this as a manual step.
+
+    Notes
+    -----
+    When a child-classes uses `cmlreaders.base_reader.MetaReader` as a meta
+    class, the reader_names dictionary stored as a class variable in
+    `cmlreaders.cmlreader.CMLReader` is updated based on the `data_types`
+    class variable in the sub-class.
+
+    """
+    def __new__(cls, name, bases, d):
+        if name is not "BaseCMLReader":
+            CMLReader.reader_names.update({x: name for x in d['data_types']})
```

Comment:
Can you do an explicit check here that `data_types` is defined and give a helpful error message if not?